### ISSUE TITLE
[Bug]: after opening a link from my browser history, each time I navigate the folder tree the "file not found" error pops up

### ISSUE THREAD

**Issue by maximelehericy** [nextcloud/server#44635]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

![image](https://github.com/nextcloud/server/assets/87175513/972c39bc-bc3c-4e3b-a1ca-e897b439d94a)

### Steps to reproduce

1. go into a folder, and a subfolder, and a subfolder
2. create an document (office in my case)
3. open the document
4. edit the document
5. close it
6. close your browser tab (or close your browser ?)
7. go to your browser history
8. click the history link to the document
9. nextcloud loads, and the document should open
10. close the document
11. navigate elsewhere in the folder tree
12. file not found error pop up should appear

### Expected behavior

there should be no file not found pop up

### Installation method

None

### Nextcloud Server version

29

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by susnux** [comment]
I can not reproduce this on 30. So maybe a 29 only issue?

This error message means that the fileid (you see in the URL like `/apps/files/123`) does not exist in the current file list.
Reasons could be:
1. The route was changed, internally the change already propagated to the `fileId` prop, but the file list is not yet loaded
2. The current view does not have that file id (this can happen on views that are not real files, like e.g. the "tags" view

We recently fixed some places that could cause both of those cases, so maybe it is already fixed?
